org-timeline.el

```diff
@@ -148,7 +145,27 @@ Return new copy of STRING."
                    (face (org-timeline--get-face)))
               (when (>= beg start-offset)
                 (push (list beg end face) tasks)))))))
-    (setq tasks (nreverse tasks))
+    (nreverse tasks)))
+
+(defun org-timeline--generate-timeline ()
+  "Generate the timeline string that will represent current agenda view."
+  (let* ((start-offset (* org-timeline-start-hour 60))
+         (current-time (+ (* 60 (string-to-number (format-time-string "%H")))
+                          (string-to-number (format-time-string "%M"))))
+         (current-offset (/ (- current-time start-offset) 10))
+         (slotline (org-timeline--add-elapsed-face
+                    "|     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |     |"
+                    current-offset))
+         (hourline (org-timeline--add-elapsed-face
+                    (let ((start-hour (abs org-timeline-start-hour)))
```

Comment:
Just curious why is the `abs` necessary?  What would be the expected use-case of setting this to `-5`?

Reply:
It is, in fact, not necessary anymore. No taking `abs` would have broken my version of the code though.

Reply:
If you really want to validate the input, this is a common pattern in elisp.

1. make a variable `(defcustom package-foo-setting ...)`
2. make a function `(defun package-foo-setting () package-foo-setting)` that just returns the value of the setting
3. use `(package-foo-setting)` to access the setting.
4. if you want to add some validation rules, you can add them to the function accessor now, for example throwing `(user-error "value must be positive number")` if the validation fails.

Reply:
Okay, thank you for that!